kyo-core/shared/src/test/scala/kyo/FiberTest.scala

```diff
@@ -730,4 +730,385 @@ class FiberTest extends Test:
+                    ()
+            }.andThen(succeed)
+        }
+
+        "quickSort" - {
+            "empty array" in {
+                val indices = Array[Int]()
+                val results = Array[AnyRef]()
+                Fiber.quickSort(indices, results, 0)
+                assert(indices.isEmpty && results.isEmpty)
+            }
+
+            "single element" in {
+                val indices = Array[Int](0)
+                val results = Array[AnyRef]("a")
+                Fiber.quickSort(indices, results, 1)
+                assert(indices.sameElements(Array(0)) && results.sameElements(Array("a")))
+            }
+
+            "partial sort with items" - {
+                "items = 0" in {
+                    val indices  = Array[Int](3, 1, 4, 2)
+                    val results  = Array[AnyRef]("c", "a", "d", "b")
+                    val original = indices.clone()
+                    Fiber.quickSort(indices, results, 0)
+                    assert(indices.sameElements(original))
+                }
+
+                "items = 1" in {
+                    val indices = Array[Int](3, 1, 4, 2)
+                    val results = Array[AnyRef]("c", "a", "d", "b")
+                    Fiber.quickSort(indices, results, 1)
+                    assert(indices(0) == 3)
+                }
+
+                "items = n-1" in {
+                    val indices = Array[Int](3, 1, 4, 2)
+                    val results = Array[AnyRef]("c", "a", "d", "b")
+                    Fiber.quickSort(indices, results, 3)
+                    assert(indices.take(3).sorted.sameElements(indices.take(3)))
+                    assert(indices(3) == 2)
+                }
+            }
+
+            "full sort" - {
+                "already sorted" in {
+                    val indices = Array[Int](0, 1, 2, 3)
+                    val results = Array[AnyRef]("a", "b", "c", "d")
+                    Fiber.quickSort(indices, results, 4)
+                    assert(indices.sameElements(Array(0, 1, 2, 3)))
+                    assert(results.sameElements(Array("a", "b", "c", "d")))
+                }
+
+                "reverse sorted" in {
+                    val indices = Array[Int](3, 2, 1, 0)
+                    val results = Array[AnyRef]("d", "c", "b", "a")
+                    Fiber.quickSort(indices, results, 4)
+                    assert(indices.sameElements(Array(0, 1, 2, 3)))
+                    assert(results.sameElements(Array("a", "b", "c", "d")))
+                }
+
+                "random order" in {
+                    val indices = Array[Int](3, 1, 4, 2)
+                    val results = Array[AnyRef]("c", "a", "d", "b")
+                    Fiber.quickSort(indices, results, 4)
+                    assert(indices.sameElements(Array(1, 2, 3, 4)))
+                    assert(results.sameElements(Array("a", "b", "c", "d")))
+                }
+            }
+
+            "can handle larger arrays" - {
+                val size = 1000
```

Comment:
`quickSort` isn't stack-safe but the implementation can handle larger arrays with a relatively low stack depth. Also, I think we can consider gathering a large number of computations as misuse.